src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -110,47 +112,89 @@ def get_comment_contents(issue_list: List[int]) -> List[PullRequestIssue]:
+        PullRequestComment.objects.create(
+            external_id=resp.body["id"],
+            pull_request_id=pullrequest_id,
+            created_at=current_time,
+            updated_at=current_time,
+            group_ids=issue_list,
+        )
+    else:
+        client.update_comment(
+            repo=repo.name, comment_id=pr_comment.external_id, data={"body": comment_body}
         )
 
-        # GitHubAppsClient (GithubClientMixin)
-        client = installation.get_client()
+        pr_comment.updated_at = timezone.now()
+        pr_comment.group_ids = issue_list
+        pr_comment.save()
+
+
+@instrumented_task(name="sentry.tasks.integrations.github_pr_comments")
+def comment_workflow(pullrequest_id: int, project_id: int):
+    gh_repo_id, pr_key, org_id, issue_list = pr_to_issue_query(pullrequest_id)[0]
+
+    try:
+        organization = Organization.objects.get_from_cache(id=org_id)
+    except Organization.DoesNotExist:
+        # TODO(cathy): release the cache key even after exceptions
+        logger.error("github.pr_comment.org_missing")
+        return
+
+    # TODO(cathy): add check for OrganizationOption for comment bot
+    if not features.has("organizations:pr-comment-bot", organization):
+        return
+
+    pr_comment = None
+    pr_comment_query = PullRequestComment.objects.filter(pull_request__id=pullrequest_id)
+    if pr_comment_query.exists():
+        pr_comment = pr_comment_query[0]
+
+    try:
+        project = Project.objects.get_from_cache(id=project_id)
+    except Project.DoesNotExist:
+        logger.error("github.pr_comment.project_missing", extra={"organization_id": org_id})
+        return
+
+    top_5_issues = get_top_5_issues_by_count(issue_list, project)
+    top_5_issue_ids = [issue["group_id"] for issue in top_5_issues]
+    issue_comment_contents = get_comment_contents(top_5_issue_ids)
+
+    try:
+        repo = Repository.objects.get(id=gh_repo_id)
+    except Repository.DoesNotExist:
+        logger.error("github.pr_comment.repo_missing", extra={"organization_id": org_id})
+        return
+
+    integration = integration_service.get_integration(integration_id=repo.integration_id)
+    if not integration:
+        logger.error("github.pr_comment.integration_missing", extra={"organization_id": org_id})
+        return
+
+    installation = integration_service.get_installation(
+        integration=integration, organization_id=org_id
+    )
+
+    # GitHubAppsClient (GithubClientMixin)
+    # TODO(cathy): create helper function to fetch client for repo
+    client = installation.get_client()
 
-        comment_body = format_comment(issue_comment_contents)
+    comment_body = format_comment(issue_comment_contents)
 
-        client.create_comment(repo=repo.name, issue_id=pr_key, data={"body": comment_body})
+    top_24_issues = issue_list[:24]  # 24 is the P99 for issues-per-PR
+    create_or_update_comment(
+        pr_comment, client, repo, pr_key, comment_body, pullrequest_id, top_24_issues
```

Comment:
can we use kwargs?

with this many, i think with this many args it gets pretty easy to mess things up when theyre ordered args